scripts/src/java/org/oppia/android/scripts/maven/GenerateMavenDependenciesList.kt

```diff
@@ -0,0 +1,546 @@
+  ).mavenDependencyList.toList()
+}
+
+/**
+ * Helper function to parse the textproto file to a proto class.
+ *
+ * @param textProtoFileName name of the textproto file to be parsed
+ * @param proto instance of the proto class
+ * @return proto class from the parsed textproto file
+ */
+private fun getProto(
+  textProtoFileName: String,
+  proto: MavenDependencyList
+): MavenDependencyList {
+  val protoBinaryFile = File("scripts/assets/$textProtoFileName")
+  val builder = proto.newBuilderForType()
+  val protoObject = FileInputStream(protoBinaryFile).use {
+    builder.mergeFrom(it)
+  }.build() as MavenDependencyList
+  return protoObject
+}
+
+fun parseArtifactName(artifactName: String): String {
+  var colonIndex = artifactName.length - 1
+  while (artifactName.isNotEmpty() && artifactName[colonIndex] != ':') {
+    colonIndex--
+  }
+  val artifactNameWithoutVersion = artifactName.substring(0, colonIndex)
+  val parsedArtifactNameBuilder = StringBuilder()
+  for (index in artifactNameWithoutVersion.indices) {
+    if (artifactNameWithoutVersion[index] == '.' || artifactNameWithoutVersion[index] == ':' ||
+      artifactNameWithoutVersion[index] == '-'
+    ) {
+      parsedArtifactNameBuilder.append(
+        '_'
+      )
+    } else {
+      parsedArtifactNameBuilder.append(artifactNameWithoutVersion[index])
+    }
+  }
+  return parsedArtifactNameBuilder.toString()
+}
+
+fun runBazelQueryCommand(rootPath: String): List<String> {
+  val rootDirectory = File(rootPath).absoluteFile
+  val bazelClient = BazelClient(rootDirectory)
+  val bazelQueryDepsNames = mutableListOf<String>()
+  val output = bazelClient.executeBazelCommand(
+    "query",
+    "\'deps(deps(//:oppia)",
+    "intersect",
+    "//third_party/...)",
+    "intersect",
+    "@maven//...\'"
+  )
+  output.forEach { dep ->
+    bazelQueryDepsNames.add(dep.substring(9, dep.length))
+  }
+  bazelQueryDepsNames.sort()
+  return bazelQueryDepsNames.toList()
+}
+
+private fun readMavenInstall(
+  pathToMavenInstall: String,
+  bazelQueryDepsNames: List<String>
+): List<MavenListDependency> {
+  val mavenInstallJson = File(pathToMavenInstall)
+  val mavenInstallJsonText = mavenInstallJson.inputStream().bufferedReader().use { it.readText() }
+  val moshi = Moshi.Builder().addLast(KotlinJsonAdapterFactory()).build()
+  val adapter = moshi.adapter(MavenListDependencyTree::class.java)
+  val dependencyTree = adapter.fromJson(mavenInstallJsonText)
+  val mavenInstallDependencyList = dependencyTree?.mavenListDependencies?.dependencyList
+  mavenInstallDependencyList?.sortBy { it -> it.coord }
+  val finalDependenciesList = mutableListOf<MavenListDependency>()
+  mavenInstallDependencyList?.forEach { dep ->
+    val artifactName = dep.coord
+    val parsedArtifactName = parseArtifactName(artifactName)
+    if (bazelQueryDepsNames.contains(parsedArtifactName)) {
+      finalDependenciesList.add(dep)
+    }
+  }
+  return finalDependenciesList
+}
+
+private fun getLicenseLinksFromPOM(
```

Comment:
This seems really complicated. Could we use an XML parser like the one from #3341 instead of manually parsing XML here?

Reply:
I think that can parse an XML file but here we don't have an XML file, we just have a string of XML file.

Reply:
Sorry, I don't understand. Below you're reading the URL of the pom file--isn't that downloading an XML file to parse?

Reply:
Actually, the link does not always point to a downloadable file - https://repo1.maven.org/maven2/com/google/protobuf/protobuf-lite/3.0.0/protobuf-lite-3.0.0.pom

Or I am getting it wrong? In this case, too, a file will get downloaded which will contain this text?

Reply:
Discussed during team chat & verified a way to use document to parse the POM file.